File: algo/src/main/java/com/hartwig/actin/algo/evaluation/FunctionCreator.java
package com.hartwig.actin.algo.evaluation;

import com.hartwig.actin.treatment.datamodel.EligibilityFunction;

import org.jetbrains.annotations.NotNull;

public interface FunctionCreator {

    @NotNull
    EvaluationFunction create(@NotNull EligibilityFunction function);
}

File: soc/src/main/kotlin/com/hartwig/actin/soc/evaluation/FunctionCreator.kt
package com.hartwig.actin.soc.evaluation

import com.hartwig.actin.treatment.datamodel.EligibilityFunction

interface FunctionCreator {

    fun create(eligibilityFunction: EligibilityFunction): EvaluationFunction

    companion object {
        operator fun invoke(createEvaluationFunction: (EligibilityFunction) -> EvaluationFunction): FunctionCreator =
            object : FunctionCreator {
                override fun create(eligibilityFunction: EligibilityFunction): EvaluationFunction =
                    createEvaluationFunction(eligibilityFunction)
            }
    }
}